cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1489,6 +1489,25 @@ func (p *BicepProvider) compileBicep(
 		return nil, fmt.Errorf("failed unmarshalling arm template from json: %w", err)
 	}
 
+	// update user-defined parameters
+	for paramKey, param := range template.Parameters {
```

Comment:
Would you mind writing a small test for `compileBicep` that mocks the call to `bicepCli.Build` so that it returns a template that has user defined types to ensure this code works?  I trust you, of course, but I don't trust myself if I ever end up refactoring this method down the line.

Reply:
test added based on typed from: https://learn.microsoft.com/azure/azure-resource-manager/bicep/user-defined-data-types